README.md

```diff
@@ -70,15 +119,279 @@ solidity: |-
+
+## Why Kafka?
+
+We selected Kafka as the first Messaging platform (and built a multi-tenant secured Kafka transport into the Kaleido platform), because Kafka has message ordering and scale characteristics that are ideally suited to the Ethereum transaction model:
+- Transactions can be sprayed across partitions, while retaining order of the transactions for a particular sender. Allowing independent and dynamic scaling of the application, kaleido-io/ethconnect bridge and Go-ethereum node components.
+- The modern replication based cloud-native and continuously available architecture is ideal for the Kaleido platform, and is likely to be a good fit for the modern Microservice architectures that are common in Blockchain projects.
+
+## Topic & Messages
+
+The topic usage is very simple:
+- One topic delivering messages into the bridge
+  - The Kafka->Ethereum bridge listens with a consumer group
+  - If configured, the Webhook->Kafka bridge sends to this topic
+  - The Kafka->Ethereum bridge updates the offset only after a message is delivered into Ethereum, and a success/failure message is sent back (at-least-once delivery)
+  - The Kafka->Ethereum bridge keeps a configurable number of messages in-flight - received from the input topic, but not yet replied to
+- One topic delivering replies
+  - All replies go to single topic, with a header that correlates replies
+  - If configured, the Webhook->Kafka bridge listens to this topic with a consumer group
+  - The Webhook->Kafka bridge marks the offset of each message after inserting into MongoDB (if the receipt store is configured)
+
+An example successful reply message:
+```json
+{
+  "headers": {
+    "id": "3eca1f95-d43a-4884-525d-8d7efa7f8c9c",
+    "requestId": "a789940d-710b-489f-477f-dc9aaa0aef77",
+    "requestOffset": "zzyly4jg5f-zze37213zm-requests:0:35479",
+    "timeElapsed": 23.160396176,
+    "timeReceived": "2018-07-25T12:15:19Z",
+    "type": "TransactionSuccess"
+  },
+  "blockHash": "0x7c6d389b27c57a0f5e4792712f1dd61a733230e3188511a6f59e0559c2c06352",
+  "blockNumber": "6329",
+  "blockNumberHex": "0x18b9",
+  "cumulativeGasUsed": "26817",
+  "cumulativeGasUsedHex": "0x68c1",
+  "from": "0xb480f96c0a3d6e9e9a263e4665a39bfa6c4d01e8",
+  "gasUsed": "26817",
+  "gasUsedHex": "0x68c1",
+  "status": "1",
+  "statusHex": "0x1",
+  "to": "0x6287111c39df2ff2aaa367f0b062f2dd86e3bcaa",
+  "transactionHash": "0xdb7578e473767105c314dd73d34b630a193227826bc6ddf6261b45a27405e7e7",
+  "transactionIndex": "0",
+  "transactionIndexHex": "0x0"
+}
+```
+
+The MongoDB receipt store adds two additional fields, used to retrieve the entries efficient on the REST interface:
+```json
+{
+  "_id": "a789940d-710b-489f-477f-dc9aaa0aef77",
+  "receivedAt": 1532520942884
+}
+```
+
+## Running the Bridge
```

Comment:
I think it may be good to cut here, transform this into a short _contributing_ or `developing` section and then move all this development information into the Wiki.

I also think this is missing a lot of detail; which can come at a later date. Documenting all the configuration parameters would be _very_ awesome.

Reply:
Or even packaging this with a `docker-compose` set up. That would make it unbelievably easy to get a local development environment (or playground) and I think it could be done.

Reply:
... think the docker-compose setup, and Docker image pushing to Dockerhub, is an awesome suggestion... but probably a separate PR. If you're interested, that would be an awesome contribution 😄